### PR TITLE
Added Quad9 analyzer and refactored DNS analyzers

### PR DESCRIPTION
**Pull request by drosetti**
### DNS analyzers have a folder hierarchy:
```
dns
|
|_____dns_malicious_detectors
|_____dns_resolvers
|_____dns_responses.py
```

Folder explainations:
* `dns_malicious_detectors`: modules in this folder have analyzers that check in the external provider databases if the domain inside the DNS query is flagged as malicious
* `dns_resolvers`: modules in this folder have analyzers that perform a DNS query and return the resolutions
* `dns_responses.py`: modules with utilities functions. Using these functions every analyzers in one folder return the same data.
    * Analyzers in `dns_malicious_detectors` return a dict: 
    ```{"observable": <observable_name>, "malicious": True/False}```
    * Analyzers in `dns_resolvers` return a dict (resolutions is empty for NXDOMAIN):
    ```
   {
       "observable": <observable_name>,
       "resolutions": [
            {
             "TTL": <int>,
             "data": <related_data>,
             "name": <domain_name>,
             "type": <DNS_query_type>
            }
       ]
    }
    ```

### Analyzers 
The old DNS analyzers in `active_dns.py` have been moved:
* `__doh_google` (function) -> `dns_resolvers/google_dns_resolver` (module)
* `__doh_cloudflare` (function) -> `dns_resolvers/cloudflare_dns_resolver` (module)
* `__doh_cloudflare_malware` (function) -> `dns_malicious_detectors/cloudflare_malicious_detector` (module)
* `__classic_dns` (function) -> `dns_resolvers/classic_dns_resolver` (module)
*  'googlesf' (module) -> `dns_malicious_detectors/googlesf` (module)

Quad9 support both malicious detector (`dns_malicious_detectors/quad9`) and DNS resolver (`dns_resolvers/quad9_dns_resolver`)

### Added support of different DNS query types
Every `dns_resolvers` can filter DNS queries with `query_type` additional param in the `analyzer_config`:
```
"additional_config_params": {
      "query_type": "A"
    }
``` 

### Modified analyzers name
The names of `active_dns` analyzer has been changed in the `analyzer_config` file and in the GUI:
* ActiveDNS_Classic -> Classic_DNS
* ActiveDNS_CloudFlare -> CloudFlare_DNS
* ActiveDNS_CloudFlare_Malware -> CloudFlare_Malicious_Detector
* ActiveDNS_Google -> Google_DNS

### CODE REVIEW

File: api_app/script_analyzers/observable_analyzers/dns/dns_malicious_detectors/quad9_malicious_detector.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from urllib.parse import urlparse
+from api_app.script_analyzers import classes
+from api_app.script_analyzers.observable_analyzers.dns.dns_responses import (
+    malicious_detector_response,
+)
+from api_app.exceptions import AnalyzerRunException
+
+
+class Quad9MaliciousDetector(classes.ObservableAnalyzer):
+    """Retrieve malicious domain in Quad9 DB.
```
Comment (mlodic) [inline api_app/script_analyzers/observable_analyzers/dns/dns_malicious_detectors/quad9_malicious_detector.py:14]: Just a reformat of the message:
```
Check if a domain is malicious by Quad9 public resolver.
Quad9 does not answer in the case a malicious domain is queried.
However, we need to perform another check to understand if that domain was blocked by the resolver or if it just does not exist.
So we perform one request to Quad9 and another one to Google.
In the case of empty response from Quad9 and a not-empty response from Google,
we can guess that the domain was in the Quad9 blacklist.
```

File: api_app/script_analyzers/observable_analyzers/dns/dns_malicious_detectors/quad9_malicious_detector.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            # if Google response, Quad9 marked the site as malicious,
+            # elsewhere the site does not exist
+            if google_answer:
+                return malicious_detector_response(self.observable_name, True)
+
+        return malicious_detector_response(self.observable_name, False)
+
+    def _quad9_dns_query(self, observable):
+        """Perform a DNS query with Quad9 service, return True if Quad9 answer the
+        DNS query.
```
Comment (mlodic) [inline api_app/script_analyzers/observable_analyzers/dns/dns_malicious_detectors/quad9_malicious_detector.py:41]: ...with a non-empty response

File: tests/utils.py [hunk 3]
```diff
@@ -185,6 +227,36 @@ def test_fortiguard(self, mock_get=None, mock_post=None):
...
+            self.job_id,
+            self.observable_name,
+            self.observable_classification,
+            {},
+        ).start()
+        self.assertEqual(report.get("success", False), True)
+
+    def test_quad9_malicious_detector(self, mock_get=None, mock_post=None):
+        report = quad9_malicious_detector.Quad9MaliciousDetector(
+            "GoogleDNSResolver",
```
Comment (mlodic) [inline tests/utils.py:252]: bad analyzer name

**Comment by Eshaan7** [comment]
To resolve the merge conflicts, remove the function you defined in `tasks.py` and add the `soft_time_limit` as an attribute to the analyzer config you made for quad9 in `analyzer_config.json`.

**Comment by drosetti** [comment]
Hi Eshaan, 

Thanks for you support, i will follow your advice and update my code.

**Comment by drosetti** [comment]
I have merged the conflicts as you suggest, I have also added "docs" folder to `.flake8` in order to ignore annoying flake8 message for docs theme import. 

I'have also modified `intel_owl.tasks.run_analyzer()`, now support nested packages like `dns`.

**Comment by mlodic** [comment]
Well done, merged! thank you!

**Pull request merged**